src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/electric/ElectricCompressor.java

```diff
@@ -0,0 +1,107 @@
+
+import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.InvUtils;
+import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.Item.CustomItem;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineHelper;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineRecipe;
+import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.api.energy.ChargableBlock;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+
+public abstract class ElectricCompressor extends AContainer {
+
+    public ElectricCompressor(Category category, ItemStack item, String id, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, id, recipeType, recipe);
+    }
+
+    @Override
+    public String getInventoryTitle() {
+        return "&9Electric Compressor";
+    }
+
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.TURTLE_HELMET);
+    }
+
+    public abstract int getSpeed();
+
+    @Override
+    protected void tick(Block b) {
+        if (isProcessing(b)) {
+            int timeleft = progress.get(b);
+            if (timeleft > 0 && getSpeed() < 10) {
+                MachineHelper.updateProgressbar(BlockStorage.getInventory(b), 22, timeleft, processing.get(b).getTicks(), getProgressBar());
+
+                if (ChargableBlock.isChargable(b)) {
+                    if (ChargableBlock.getCharge(b) < getEnergyConsumption()) return;
+                    ChargableBlock.addCharge(b, -getEnergyConsumption());
+                    progress.put(b, timeleft - 1);
+                }
+                else progress.put(b, timeleft - 1);
+            }
+            else if (ChargableBlock.isChargable(b)) {
+                if (ChargableBlock.getCharge(b) < getEnergyConsumption()) return;
+                ChargableBlock.addCharge(b, -getEnergyConsumption());
+
+                BlockStorage.getInventory(b).replaceExistingItem(22, new CustomItem(new ItemStack(Material.BLACK_STAINED_GLASS_PANE), " "));
+                pushItems(b, processing.get(b).getOutput());
+
+                progress.remove(b);
+                processing.remove(b);
+            }
+        }
+        else {
+            for (int slot: getInputSlots()) {
+                BlockMenu inv = BlockStorage.getInventory(b);
+                ItemStack input = inv.getItemInSlot(slot);
+
+                if (input == null || input.getType() == Material.AIR || input.getAmount() == 0) continue;
+                int amount = input.getAmount();
+
+                MachineRecipe r = null;
+                if (SlimefunManager.isItemSimiliar(input, SlimefunItems.REINFORCED_ALLOY_INGOT, true)) {
```

Comment:
This is not a good idea.
You should copy over the actual recipes from the Compressor instead of this hacky workaround.

Reply:
But Compressor also has carbon and compressed carbon recioes which are included in Carbon Press. If I do it like that then Carbon Press will be redundant.

Reply:
Then filter those out, simple as that.

Reply:
Uhm the recipes of multiblocks are just a mess.

Reply:
Not really. In MiscSetup is already the same implementation you need, just for the Ore Grinder...

Reply:
Also I have different speeds for each material so I can't really change this.